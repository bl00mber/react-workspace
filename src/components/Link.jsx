import React from 'react';
import PropTypes from 'prop-types';

const Link = ({ active, children, onClick }) => (
  <button
    disabled={active}
    style={{
      textDecoration: active ? 'underline' : undefined,
    }}
    href=""
    onClick={(e) => {
      e.preventDefault();
      onClick();
    }}
  >
    {children}
  </button>
);

Link.propTypes = {
  active: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Link;
