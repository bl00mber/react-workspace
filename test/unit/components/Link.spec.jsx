import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Link from '@/components/Link';

describe('Link', () => {
  test('should match enzyme shallow snapshot', () => {
    const props = {
      active: false,
      onClick: jest.fn(),
    };
    const wrapper = shallow(<Link {...props}>Test 1</Link>);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({
      active: true,
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should call onClick clicking on link', () => {
    const props = {
      active: false,
      onClick: jest.fn(),
    };
    const wrapper = shallow(<Link {...props}>Test 1</Link>);
    const clickEvent = {
      preventDefault: jest.fn(),
    };
    wrapper.find('button').simulate('click', clickEvent);
    expect(props.onClick).toHaveBeenCalledWith();
    expect(clickEvent.preventDefault).toHaveBeenCalledWith();
  });
});
