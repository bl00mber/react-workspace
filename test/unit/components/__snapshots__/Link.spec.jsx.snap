// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should match enzyme shallow snapshot 1`] = `
<a
  href=""
  onClick={[Function]}
>
  Test 1
</a>
`;

exports[`Link should match enzyme shallow snapshot 2`] = `
<span>
  Test 1
</span>
`;
